zerver/views/streams.py

```diff
@@ -455,6 +455,11 @@ def update_subscription_properties_backend(request, user_profile, subscription_d
         if sub is None:
             return json_error(_("Not subscribed to stream id %d") % (stream_id,))
 
+        if property_converters[property] == check_bool:
+            if value == 'false':
+                value = False
+            elif value == 'true':
+                value = True
```

Comment:
That didn't seem right when I was doing it -- I think I know where my issue was now, though; I'll update this PR tomorrow and remove this piece.